sql/sys_vars.cc

```diff
@@ -4208,14 +4208,23 @@ static bool check_tx_isolation(sys_var *self, THD *thd, set_var *var)
 
 // NO_CMD_LINE - different name of the option
 static Sys_var_tx_isolation Sys_tx_isolation(
-       "tx_isolation", "Default transaction isolation level",
+       "tx_isolation", "Default transaction isolation level."
+       "This variable is deprecated and will be removed in a future release.",
+       NO_SET_STMT SESSION_VAR(tx_isolation), NO_CMD_LINE,
+       tx_isolation_names, DEFAULT(ISO_REPEATABLE_READ),
+       NO_MUTEX_GUARD, NOT_IN_BINLOG, ON_CHECK(check_tx_isolation),
+       ON_UPDATE(0), DEPRECATED("'@@transaction_isolation'"));
+
+// NO_CMD_LINE
+static Sys_var_tx_isolation Sys_transaction_isolation(
+       "transaction_isolation", "Default transaction isolation level",
        NO_SET_STMT SESSION_VAR(tx_isolation), NO_CMD_LINE,
```

Comment:
Here I think we can omit `NO_SET_STMT`.
Instead of `NO_CMD_LINE` we should write `CMD_LINE(REQUIRED_ARG)`. 
I guess the same should be done for `tx_read_only`. 
@FooBarrior  confirm please.

Reply:
>Here I think we can omit NO_SET_STMT
If so, it needs to be covered with tests

>we should write CMD_LINE(REQUIRED_ARG)
seems correct, but what's the behavioral difference?

Reply:
Currently, the command line option is managed separately in `sql/mysqld`, so `NO_CMD_LINE` here. Changing to `CMD_LINE(REQUIRED_ARG)` is fine, but we have to delete the entry in `my_long_options` in `sql/mysqld`.

Reply:
Well, according to how i figured it out, it seems to have no big difference on practice: `my_long_options` are parsed just together with sys_var options (`get_options` in `mysqld.cc`). I also see that we have a practice of leaving the option in `my_long_options` while having a non-cmdline twink in `all_sys_vars`. 

For example, `default-storage-engine` exists in `my_long_options`, while having `default_storage_engine` in `sys_vars.cc`. And it has `NO_CMD_LINE`. 

So let's follow this convention as well. This means -- leave `NO_CMD_LINE`, as it's now.

Reply:
agree, `NO_SET_STMT` can be omitted, but please add test for `SET STATEMENT` https://mariadb.com/kb/en/set-statement/